src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Azure.Quantum;
+using Microsoft.Azure.Quantum.Exceptions;
+using static Microsoft.Quantum.EntryPointDriver.Driver;
+using Microsoft.Quantum.EntryPointDriver.Mock;
+using Microsoft.Quantum.Runtime;
+using Microsoft.Quantum.Runtime.Submitters;
+using Microsoft.Quantum.Simulation.Common.Exceptions;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Quantum.EntryPointDriver.Azure
+{
+    using Environment = System.Environment;
+
+    /// <summary>
+    /// Provides entry point submission to Azure Quantum.
+    /// </summary>
+    public static class Azure
+    {
+        /// <summary>
+        /// Submits an entry point to Azure Quantum. If <paramref name="qirSubmission"/> is non-null and a QIR submitter
```

Comment:
How are `settings` populated? How is it determined if a QIR submitter is available for the target?

Reply:
`AzureSettings` is created by System.CommandLine using some reflection magic that sets the properties based on comparing the property name to the command line option names.

https://github.com/microsoft/qsharp-runtime/blob/7cff67b6bbcc6ebc1509b634b1dbceb21a62c6db/src/Simulation/EntryPointDriver/Driver.cs#L409-L412

The availability of a QIR submitter is determined by `Azure.QirSubmitter` returning a non-null value, which will be determined by the return value of the QIR submitter factory once that is implemented. That's what the chain of `if`s here do - they try each submitter/machine type in order until one returns non-null.

https://github.com/microsoft/qsharp-runtime/blob/7cff67b6bbcc6ebc1509b634b1dbceb21a62c6db/src/Simulation/EntryPointDriver/Azure/Azure.cs#L39-L55

Reply:
Thanks for the explanation Sarah!